compiler/rustc_trait_selection/src/traits/mod.rs

```diff
@@ -219,6 +220,7 @@ fn do_normalize_predicates<'tcx>(
     cause: ObligationCause<'tcx>,
     elaborated_env: ty::ParamEnv<'tcx>,
     predicates: Vec<ty::Predicate<'tcx>>,
+    outlives: bool,
```

Comment:
Could you make the parameter name more explicit? I don't understand what this toggles.

Reply:
Okay. In `param_env`, normalization will be called twice according to whether it is a predicates of type `TypeOutlives`. This variable is used to distinguish between the two calls.